front: re-enable dependabot for direct dependencies

emersion opened the pull request:
In #2969, dependabot got disabled because it was too spammy. However, disabling dependency updates makes us fall behind pretty hard and makes version upgrades quite painful. Instead of gradually integrating new dependency upgrades and finding bugs as they arise, we're forced to make a huge commit upgrading many packages at once. Regressions are harder to narrow down and humans have to remember and volunteer to perform this unrewarding upgrade task.

As a middle ground, re-enable dependabot updates for direct dependencies only, and group all updates in a single weekly PR. This should make it less spammy than enabling it for all dependencies (transitive dependencies included), while at the same time allowing us to keep up with new versions of libraries we're using.

anisometropie commented:
isn’t there a way to batch all the bumps into one single PR, like every week one big bump PR with all the new cool stuff?

woshilapin reviewed (approved):
How about also changing the frequency to "weekly" instead of "daily", this might also help with the "too noisy" problem and still be acceptable?

emersion commented:
Switched to weekly updates grouped in a single PR.

emersion commented:
If it's a single weekly PR, maybe we can re-enable updates for transitive dependencies.

anisometropie commented:
> If it's a single weekly PR, maybe we can re-enable updates for transitive dependencies.

I’d say go for it

emersion committed:
front: re-enable dependabot

In #2969, dependabot got disabled because it was too spammy.
However, disabling dependency updates makes us fall behind pretty
hard and makes version upgrades quite painful. Instead of gradually
integrating new dependency upgrades and finding bugs as they arise,
we're forced to make a huge commit upgrading many packages at once.
Regressions are harder to narrow down and humans have to remember
and volunteer to perform this unrewarding upgrade task.

As a middle ground, re-enable dependabot but group all updates in a
single weekly PR. This should make it less spammy than getting one
PR per dependency upgrade (transitive dependencies included), while
at the same time allowing us to keep up with new versions of
libraries we're using.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Alrighty
